com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/WorkspaceServiceTest.java

```diff
@@ -15,31 +15,113 @@
  */
 package com.ca.lsp.cobol.service;
 
+import com.broadcom.lsp.domain.cobol.databus.api.DataBusBroker;
 import com.broadcom.lsp.domain.cobol.databus.impl.DefaultDataBusBroker;
+import com.broadcom.lsp.domain.cobol.event.model.RequiredCopybookEvent;
 import com.broadcom.lsp.domain.cobol.event.model.RunAnalysisEvent;
+import com.google.gson.JsonPrimitive;
 import lombok.extern.slf4j.Slf4j;
 import org.eclipse.lsp4j.DidChangeWatchedFilesParams;
-import org.eclipse.lsp4j.FileChangeType;
+import org.eclipse.lsp4j.ExecuteCommandParams;
 import org.eclipse.lsp4j.FileEvent;
 import org.junit.Test;
 import org.mockito.ArgumentCaptor;
 
-import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
-import static junit.framework.TestCase.assertNotNull;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
+import static com.ca.lsp.cobol.service.delegates.validations.UseCaseUtils.DOCUMENT_URI;
+import static com.ca.lsp.core.cobol.model.ErrorCode.MISSING_COPYBOOK;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.eclipse.lsp4j.FileChangeType.Changed;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.*;
 
+/**
+ * This test checks the entry points of the {@link org.eclipse.lsp4j.services.WorkspaceService}
+ * implementation.
+ */
 @Slf4j
 public class WorkspaceServiceTest {
+
+  /**
+   * Test of the workspace/executeCommand entry point. Assert that on a MISSING_COPYBOOK the {@link
+   * RequiredCopybookEvent} is fired.
+   */
+  @Test
+  public void testExecuteCommand() {
+    DataBusBroker broker = mock(DataBusBroker.class);
+    ArgumentCaptor<RequiredCopybookEvent> captor = forClass(RequiredCopybookEvent.class);
+    String copybookName = "COPYBOOK";
+
+    CobolWorkspaceService service = new CobolWorkspaceServiceImpl(broker, null);
+
+    CompletableFuture<Object> result =
+        service.executeCommand(
+            new ExecuteCommandParams(
+                MISSING_COPYBOOK.name(),
+                asList(new JsonPrimitive(copybookName), new JsonPrimitive(DOCUMENT_URI))));
```

Comment:
I didn't get why is required to use ```JsonPrimitive``` object, is not used in every definition of ```ExecuteCommandParams```

Reply:
This `ExecuteCommandParams` has the `List<Object> arguments` field, but it uses not java objects, but JSONs, that are represented in java as `JsonObject` or, in case of strings, as `JsonPrimitive`